### PR TITLE
feat(js): Include headers and tabs in separate components

### PR DESCRIPTION
**Pull request by desiprisg**
### What changed? Why was the change needed?

We decided to include the whole content of the pages shown in the inbox for the separate components. Also a one line change is possible by uncommenting `InboxContent` to allow mounting the whole content (with routing) for usage in custom popover.
### Screenshots

Screenshot 2024-08-14 at 5 17 51 PM
Screenshot 2024-08-14 at 5 18 08 PM

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/js/src/ui/components/Inbox.tsx [hunk 1]
```diff
@@ -21,28 +21,40 @@ export type InboxProps = {
...
 };
 
-const InboxContent = (props: InboxContentProps) => {
-  const [currentScreen, setCurrentScreen] = createSignal<Screen>(Screen.Inbox);
+export const InboxContent = (props: InboxContentProps) => {
+  const [currentPage, setCurrentPage] = createSignal<InboxPage>(props.initialPage || InboxPage.Notifications);
   const { tabs, filter } = useInboxContext();
   const style = useStyle();
 
+  const navigateToPage = createMemo(() => (page: InboxPage) => {
```
Comment (BiswaViraj) [inline packages/js/src/ui/components/Inbox.tsx:43]: The extra arrow function can be removed. Is there a reason why it needs to return a function ?

```
const navigateToPage = createMemo(page => {
  if(props.hideNav) {
    return
  }
  setCurrentPage(page)
})
```

File: packages/js/src/ui/components/elements/Header/PreferencesHeader.tsx [hunk 2]
```diff
@@ -12,15 +13,19 @@ export const PreferencesHeader = (props: PreferencesHeaderProps) => {
...
-          'preferencesHeader__back__button',
-          'nt-h-6 nt-w-6 nt-flex nt-justify-center nt-items-center nt-rounded-md nt-relative hover:nt-bg-foreground-alpha-50 focus:nt-bg-foreground-alpha-50 nt-text-foreground-alpha-600'
+      <Show when={props.navigateToNotifications}>
+        {(navigateToNotifications) => (
+          <button
+            class={style(
+              'preferencesHeader__back__button',
+              'nt-h-6 nt-w-6 nt-flex nt-justify-center nt-items-center nt-rounded-md nt-relative hover:nt-bg-foreground-alpha-50 focus:nt-bg-foreground-alpha-50 nt-text-foreground-alpha-600'
+            )}
+            onClick={navigateToNotifications()}
```
Comment (LetItRock) [inline packages/js/src/ui/components/elements/Header/PreferencesHeader.tsx:23]:
```suggestion
            onClick={navigateToNotifications}
```
❓ **question:** should it be like that?

**Review by LetItRock — APPROVED** [review]
looks good to me, just a two small suggestions from me and Biswa ;)

File: packages/js/src/ui/components/Inbox.tsx (hunk 1 above)
Comment (desiprisg) [reply packages/js/src/ui/components/Inbox.tsx:43]: Yes, it's the equivalent of `useCallback` in react.
Essntially `useCallback` is a `useMemo` with an extra arrow function.

File: packages/js/src/ui/components/elements/Header/PreferencesHeader.tsx (hunk 2 above)
Comment (desiprisg) [reply packages/js/src/ui/components/elements/Header/PreferencesHeader.tsx:23]: If we use `createMemo`, I believe so.

**Pull request merged**